test/truefi2/lines-of-credit/CreditModel.test.ts

```diff
@@ -363,10 +384,59 @@ describe('CreditModel', () => {
         [1000, 40, 100],
       ].map(([collateral, ltvRatio, result]) =>
         it(`when ${collateral} TRU is staked with ltvRatio=${ltvRatio}%, borrow limit rises by up to $${result}`, async () => {
-          await creditModel.setBorrowLimitConfig(0, 0, 0, 0, ltvRatio * 100)
+          await creditModel.setStakingConfig(ltvRatio * 100, 0)
           expect(await creditModel.conservativeCollateralValue(mockPool.address, parseTRU(collateral))).to.equal(parseEth(result))
         }))
     })
+
+    describe('conservativeCollateralRatio', () => {
+      const collateral = 1000
+
+      beforeEach(async () => {
+        const ltvRatio = 40
+        await creditModel.setStakingConfig(ltvRatio * 100, 0)
+      })
+
+      ;[
+        [1000, 10],
+        [500, 20],
+        [250, 40],
+        [200, 50],
+        [100, 100],
+        [75, 100],
+        [0, 0],
+      ].map(([borrowed, result]) =>
+        it(`when borrowed amount is ${borrowed} collateral ratio is at ${result}%`, async () => {
+          expect(await creditModel.conservativeCollateralRatio(mockPool.address, parseTRU(collateral), parseEth(borrowed)))
+            .to.equal(result * 100)
+        }))
+    })
+
+    describe('effectiveScore', () => {
+      const collateral = 1000
+      const borrowedAmount = 250
+
+      beforeEach(async () => {
+        const ltvRatio = 40
+        const effectiveScorePower = 2
+        await creditModel.setStakingConfig(ltvRatio * 100, effectiveScorePower)
+      })
+
+      ;[
+        [255, 255],
+        [223, 228],
+        [191, 201],
+        [159, 174],
+        [127, 147],
+        [95, 120],
+        [63, 93],
+        [31, 66],
+      ].map(([score, effectiveScore]) =>
+        it(`staking ${collateral} TRU increases
+        score ${score} to ${effectiveScore}`, async () => {
+          expect(await creditModel.effectiveScore(score, mockPool.address, parseTRU(collateral), parseEth(borrowedAmount))).to.eq(effectiveScore)
+        }))
+    })
```

Comment:
Missing tests with collateral=0, with pool with different decimal count, with collateral below borrowed amount and with different score power